bftengine/src/bftengine/ClientsManager.hpp

```diff
@@ -244,15 +248,22 @@ class ClientsManager : public ResPagesClient<ClientsManager>, public IPendingReq
 
   class RepliesInfo {
    public:
-    void deleteOldestReplyIfNeededSafe(NodeIdType clientId, uint16_t maxNumOfReqsPerClient);
-    bool insertOrAssignSafe(ReqId reqSeqNum, Time time);
+    void deleteReplyIfNeededSafe(NodeIdType clientId,
+                                 ReqId reqSeqNum,
+                                 uint16_t maxNumOfReqsPerClient,
+                                 uint16_t reqOffset);
+    void insertOrAssignSafe(NodeIdType clientId,
+                            ReqId reqSeqNum,
+                            uint16_t maxNumOfReqsPerClient,
+                            uint16_t requestOffset);
     bool findSafe(ReqId reqSeqNum);
+    uint16_t getOffsetSafe(ReqId reqSeqNum);
 
     bool find(ReqId reqSeqNum) const;
 
    public:
     std::mutex repliesMapMutex_;
-    std::map<ReqId, Time> repliesMap_;
+    std::map<ReqId, uint16_t> repliesMap_;
```

Comment:
See my comment about offset, why using. uint16_t and not uint32_t or better size_t ?

Reply:
uint16_t should be good enough, see my comment above.